### DISCUSSION TITLE
QC based on branch length

### DISCUSSION THREAD

**Discussion by jeromekelleher** [jeromekelleher/sc2ts-paper#219] [General]
Looking at the branch length distribution is a use QC tool I think, and should give us some help in identifying how much of an effect time travellers are having. The basic intuition is, if we get a sequence that time travels back 18 months, we should have a lot of 18 month long branches, when the ARG finally catches up.

First up, let's look at the distribution of all branch (or more accurately edge) lengths in the ARG:

![all-branches](https://github.com/user-attachments/assets/ac9b29ac-c6c7-42bc-9520-bdaa3181c818)

Note that I've filtered out edges < 1 day, because these will all be caused by our intentionally simplistic methods to assigning lengths to branches of nodes added through UPGMA tree building and parsimony hueristics.

On a log-y scale, it looks like we have a fairly clean exponential fall off up until about 150 days:

![all-branches-log](https://github.com/user-attachments/assets/2b188001-4468-4ffc-95e5-1e48370122a8)

**Comment by jeromekelleher** [comment]
Next, let's look at the distribution of branch lengths of exact matches:

![length-exact-match](https://github.com/user-attachments/assets/f13e9c07-6bc5-45e5-ad10-cc84d3c9a9f7)

You would imagine that the probability of observing an exact match after a long time should be very small, so I thought this could be a good metric to spot time travellers with. Here, we seem to have a very clear exponential drop off (log scale) with a big bump around 150 days, and a clear excess at longer lengths.

To me this is evidence of a time travel problem.

**Reply by hyanwong** [reply]
The exact-match branch length is a great idea. I agree with this analysis. Can we break it down into specific nodes in any way? I guess that's what you have done in the other thread in this discussion. Thanks for following up on this.

**Comment by jeromekelleher** [comment]
Looking at it in a different way, let's try to identify long branches of large effect, i.e. those that end up subtending large numbers samples (these will also correspond to major new clades/saltations, so they won't always be a sign of problems).

![Screenshot from 2024-09-21 23-03-45](https://github.com/user-attachments/assets/ce147faf-c225-404a-9524-904a92d0594c)

Here's some of the nodes identified as outliers

**Reply by jeromekelleher** [reply]
There's definitely some dodginess going on here. The recombinants are probably iffy, put let's put those to one side for now.

Node 341373 has a branch length of 116 days, and also has 19 exact match children with a mean branch length of 182 days. This seems wildly implausible. However, time travel can also go the other direction, so perhaps these samples were added in 180 days too late? We'd need to look at the grandchildren I guess to get a better sense of how far the bulk of close-matches are from this node.

![Screenshot from 2024-09-21 23-38-13](https://github.com/user-attachments/assets/c164ec95-cc5b-4fb3-a0b2-c79f68074c9c)

**Comment by jeromekelleher** [comment]
I ran the long ARG inference with a wider retrospective interval (60 days) until 2021-10-01, and to me the results look substantially worse:
![Screenshot from 2024-09-22 23-13-10](https://github.com/user-attachments/assets/f9b0b4ae-a9a9-4ace-b022-89cb42c75ae2)

Now I haven't actually compared date-for-date here, so maybe it's not a fair comparison. But, it's clearly not good, so I think that it's showing that we need to make the retrospective window narrow, not wide. It doesn't matter so much if we lose a few days of data at  the start of the outbreak, but it's vital that we add big new things in the right time order.

I'm running again with a retrospective window of 20 and a required group size of 100. Let's see what happens.

**Reply by jeromekelleher** [reply]
Plotting function added to the TreeInfo class  here: https://github.com/jeromekelleher/sc2ts/pull/289

**Reply by hyanwong** [reply]
Great. This was a good thing to do. Thanks. I agree about shortening the window. It appears (as you said) that it is more important to weed out time travellers, which can badly affect the inferred topology, rather than focus on getting the earliest representatives of a new outbreak.
